core/domain/feedback_services.py

```diff
@@ -291,12 +308,39 @@ def _add_feedback_message_reference(user_id, reference):
         enqueue_feedback_message_email_task(user_id)
 
 
-def send_feedback_message_email(exploration_id, thread_id, message_id):
+def update_feedback_email_retries(user_id):
+    model = feedback_models.UnsentFeedbackEmailModel.get(user_id)
+    time = (datetime.datetime.utcnow() - model.created_on).seconds
+
+    if time > feconf.DEFAULT_FEEDBACK_MESSAGE_EMAIL_COUNTDOWN_SECS:
+        model.retries += 1
+        model.put()
+
+
+def delete_feedback_message_references(user_id, references_length):
+    model = feedback_models.UnsentFeedbackEmailModel.get(user_id)
+
+    if references_length == len(model.feedback_message_references):
+        model.delete()
+    else:
+        message_references = (
+            model.feedback_message_references[references_length:])
+        model.delete()
+        model = feedback_models.UnsentFeedbackEmailModel(
+            id=user_id,
+            feedback_message_references=message_references)
+        model.put()
+        enqueue_feedback_message_email_task(user_id)
+
+
+def send_feedback_message_email(author_id, exploration_id, thread_id,
```

Comment:
Yeah it actually does not send the email.
How about `add_feedback_message_email` rather then this one ?